server/src/routes/user.js

```diff
@@ -95,4 +95,36 @@ router.post('/:username/posts', [upload.single('image'), UserValidation.addPost,
     }
 }]);
 
+// @route    PUT users/board/:id
+// @desc     Update posts in a board
+// @access   Private
+router.put('/board/:id', async (req,res) => {
+    try {
+        const board = await Board.findById(req.params.id);
+        if(board.user.toString() !== req.decoded._id.toString()) {
+            return res.status(404).json({msg: 'You do not have the authorization to add to this board'});
+        }
+        // Check to see if there is a board with that id
+        if(!board) {
+            return res.status(404).json({msg: 'Board not found'});
+        }
+        // Check to see if post has already been added to board
+        if (
+            board.posts.filter(post => post.toString() === req.body._id).length > 0
```

Comment:
You can perform an addToSet to make sure the post isn't already there:
https://stackoverflow.com/questions/17752737/how-to-push-a-new-document-in-mongodb-but-only-if-it-does-not-exist/17753083